src/classes/RD2_ElevateInformation_CTRL.cls

```diff
@@ -0,0 +1,172 @@
+        get {
+            if (finder == null) {
+                finder = new RecordFinder();
+            }
+            return finder;
+        }
+        set;
+    }
+
+    /**
+    * @description Returns data required for the Elevate Information widget
+    * @param recordId Recurring Donation Id
+    * @return Map<String, Object> Data value by key
+    */
+    @AuraEnabled(cacheable=false)
+    public static Map<String, Object> getData(Id recordId) {
+        Map<String, Object> data = new Map<String, Object>();
+
+        try {
+            data = new Map<String, Object> {
+                'isElevateCustomer' => RD2_ElevateIntegrationService.isElevateCreateEnabled(),
+                'alert' => RD2_VisualizeScheduleController.getValidationAlert(recordId),
+                'hasFieldPermissions' => hasFieldPermissions(),
+                'errorMessage' => getLatestErrorMessage(recordId)
+            };
+
+        } catch (Exception e) {
+            UTIL_AuraEnabledCommon.throwAuraHandledException(e.getMessage());
+        }
+
+        return data;
+    }
+
+    /**
+     * @description Checks if the user has access for fields retrieved by the Elevate information widget
+     * @return Boolean
+     */
+    private static Boolean hasFieldPermissions() {
+        Set<String> requiredFields = new Set<String>{
+            UTIL_Namespace.StrTokenNSPrefix('Status__c'),
+            UTIL_Namespace.StrTokenNSPrefix('ClosedReason__c'),
+            UTIL_Namespace.StrTokenNSPrefix('CommitmentId__c')
+        };
+
+        Boolean hasPermissions = true;
+
+        for (String fld : requiredFields) {
+            DescribeFieldResult dfr = UTIL_Describe.getFieldDescribe('npe03__Recurring_Donation__c', fld);
+            if (!dfr.isAccessible()) {
+                hasPermissions = false;
+                break;
+            }
+        }
+
+        return hasPermissions;
+    }
+
+    /***
+     * @description Displays the latest error when latest payment could not be charged or
+     * any other issue with the recurring commitment in Elevate
+     * @param recordId Recurring Donation Id
+     * @return String The latest error message if no successful payment is created after it.
+     */
+    private static String getLatestErrorMessage(Id recordId) {
+        Error__c error = finder.getLatestError(recordId);
+
+        if (error == null) {
+            return null;
+        }
+
+        //Do not return error if the error is outdated,
+        //ie created before the latest successful payment
+        Opportunity paidOpp = finder.getLatestPaidOpportunity(recordId);
+
+        Boolean isLatestError = paidOpp == null
+            || paidOpp.CreatedDate <= error.CreatedDate;
+
+        return isLatestError
+            ? String.format(System.Label.commonErrorPrefix, new String[]{ error.Full_Message__c })
+            : null;
+    }
+
+
+    /***
+     * @description Retrieves records
+     */
+    public without sharing class RecordFinder {
```

Comment:
I need a new inner class with "without sharing" property. Also, during dev, I planned to develop a mock class (needs to implement an interface) that will return a mock Error or Opportunity since CreatedDate differentiation does not account for ms, just seconds which is difficult to reproduce in tests. Also, this controller and the inner class can be used to retrieve error records when the user clicks on the "View Error Log" (W-8211933) so one Apex class can be enabled for both Elevate Information and Error Log components.
Feel free to suggest another class name if the `RecordFinder` does not work (we have a similar `UTIL_Finder` class that builds a SOQL/SOSL and returns records.

Reply:
Sounds reasonable.